### DISCUSSION TITLE
Looking for Maintainers

### DISCUSSION THREAD

**Discussion by souzatharsis** [souzatharsis/podcastfy#212] [Announcements]
Hi all,

It's just incredible how this repo has grown in just 2 months!
With growth comes great responsibilities.

I am currently the sole maintainer of the package and clearly cannot keep up with the daily requests.

I would love to onboard volunteers who would be excited to become main maintainers going forward.
If you are interested please reach out. I'd be happy to help you take from here.

I am looking to complete this process by the end of January.

Best,
Tharsis.

**Comment by Nibezak** [comment]
Hey souzatharsis ... I would be happy to help, how do I sign up

**Comment by souzatharsis** [comment]
You just did!

I'll reach out in a couple of weeks.

Anyone can join. I'll do my best to prepare documentation to on-board you.

Thanks for your interest!

**Reply by Nibezak** [reply]
still waiting for the sign up button

**Comment by aidivas** [comment]
Hey I’m happy to help…although I’m not a true coder but other tech skills

**Comment by souzatharsis** [comment]
Hey @Nibezak @aidivas 

Thanks for your interest!

What kind of information, instructions would you need to get started as a maintainer of this project?

I am ready to give you the credentials and would like to know how I can best onboard you.

Best,
Tharsis

**Reply by aidivas** [reply]
Hello

Look forward to helping out. I’m currently working my way through all the features and your md file guides via a python CLI setup…there’s a lot of cool features in those files..

What tasks/help are you looking for?
What does a maintainer usually do?

**Comment by souzatharsis** [comment]
Hi, I've added @caiquecober as maintainer and he now has permissions to add others as well as approve and merge PRs.

**Comment by aidivas** [comment]
@caiquecober hello. I see you added a beta fast api URL…any info on how to run this beta version. I was about to try and setup my own fastapi using the original api files..thanls

**Comment by caiquecober** [comment]
Hey aldiva a added a small section to the read me

Containerize podcastify and launch the api Dockerfile_api

 Make requests to the api look at the notebook for a clear example
fetch_audio(request_data, ENDPOINT, BASE_URL)

If you want to run via pip you can run it via launching the api after
downloading the package

uvicorn podcastify.api.fast_app:app --host 0.0.0.0 --port 8000

**Reply by aidivas** [reply]
@caiquecober hey thanks so much for the hard work here! It was on my weekend project list!! 

 I have tested it locally on w10 using vscode so far. Will deploy to Linux behind a reverse proxy and update these notes next. Will do a more detailed doc then.
 
Im adding a few notes for windows local testers using CURL and docker

there was a JSON issue with this line - the don't (apostrophe) word breaks the curl JSON - if you change it to do not its fine. Im sure this works in the python notebook...

"engagement_techniques": ["questions", "examples", "analogies", "humor"],
    "user_instructions": **"Don’t use the word** 'Delve'; use variants of exactly; focus on storytelling of what and when role2 speaks it should be in the first person",
    "output_language": "English"
}

To find the api outputs if you test with Curl  locally - its here
GET /audio/{filename}
• Description: Serves the generated audio file from the temporary directory.
• Example:
  curl [http://localhost:8000/audio/podcast_XXXXXXXX.mp3]
• Retrieves the audio file for playing or downloading.
 
 Below is a summary of the available REST API endpoints in the Podcastfy FastAPI app. These endpoints are exposed on the container’s port (for example, 8000 when running via Docker).

1. GET /health  
   • Description: Checks if the API is running.  
   • Example:  
  curl http://localhost:8000/health  
   • Returns: JSON like {"status": "healthy"}

2. POST /generate  
   • Description: Accepts a JSON payload with podcast parameters (e.g. API keys, URLs to process, TTS model, conversation options, etc.) and kicks off podcast generation.  
   • Required parameters (in the payload):  
  - generate_podcast (boolean)  
  - google_key (string)  
  - openai_key (string)  
  - elevenlabs_key (string)  
  - urls (list of URLs)  
  - name, tagline (strings)  
  - creativity (number)  
  - conversation_style (list of strings)  
  - roles_person1, roles_person2 (strings)  
  - dialogue_structure (list of strings)  
  - tts_model (string)  
  - is_long_form (boolean)  
  - engagement_techniques (list of strings)  
  - user_instructions (string)  
  - output_language (string)  
   • Example curl command:  
   
   To change the default fast api port - you can do this in the fast_app.py script here
   
   f __name__ == "__main__":
    host = os.getenv("HOST", "127.0.0.1")
    port = int(os.getenv("PORT", 8001))  # Changed default port from 8080 to 8001
    uvicorn.run(app, host=host, port=port)

**Comment by aliasgarsogiawala** [comment]
hey @souzatharsis , I know im a lil late for this request but i'd love to be of help
